Sources/MongoSwift/MongoCursor.swift

```diff
@@ -197,37 +192,65 @@ public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
      *   - `DecodingError` if an error occurs decoding the server's response.
      */
     public func nextOrError() throws -> T? {
-        if let next = self.next() {
-            return next
-        }
-        if let error = self.error {
-            throw error
+        let next = self.next()
+        if let next = next {
+            switch next {
+            case let .success(doc):
+                return doc
+            case let .failure(error):
+                throw error
+            }
         }
         return nil
     }
 
-    /// Returns the next `Document` in this cursor, or `nil`. After this function returns `nil`, the caller should use
-    /// the `.error` property to check for errors. For tailable cursors, users should also check `isAlive` after this
-    /// method returns `nil`, to determine if the cursor has the potential to return any more data in the future.
-    public func next() -> T? {
+    /**
+     * Returns an array of type  `Document` from this cursor, or throws
```

Comment:
since this is a cursor over a generic `Codable` type `T`, this may not actually be an array of documents.